velox/dwio/dwrf/writer/Writer.cpp

```diff
@@ -844,20 +845,18 @@ dwrf::WriterOptions getDwrfOptions(const dwio::common::WriterOptions& options) {
         std::to_string(options.zstdCompressionLevel.value()));
   }
 
-  dwrf::WriterOptions dwrfOptions;
-  dwrfOptions.config = Config::fromMap(configs);
-  dwrfOptions.schema = options.schema;
-  dwrfOptions.memoryPool = options.memoryPool;
-  dwrfOptions.spillConfig = options.spillConfig;
-  dwrfOptions.nonReclaimableSection = options.nonReclaimableSection;
-  return dwrfOptions;
+  return configs;
 }
 
 std::unique_ptr<dwio::common::Writer> DwrfWriterFactory::createWriter(
     std::unique_ptr<dwio::common::FileSink> sink,
     const std::shared_ptr<dwio::common::WriterOptions>& options) {
-  auto dwrfOptions = getDwrfOptions(*options);
-  return std::make_unique<Writer>(std::move(sink), dwrfOptions);
+  auto configs = getDwrfOptionsConfigs(*options);
+  auto dwrfOptions = std::dynamic_pointer_cast<dwrf::WriterOptions>(options);
+  dwrfOptions->config = Config::fromMap(configs);
+  VELOX_CHECK_NOT_NULL(
```

Comment:
Should we check if dwrfOptions is NULL before accessing it?

Reply:
@rui-mo Good catch. Move https://github.com/facebookincubator/velox/pull/8864/files#diff-c00f448770f041f78addaa96e1ef6954732751b57b001950e92865a2f3d830e2R857 after https://github.com/facebookincubator/velox/pull/8864/files#diff-c00f448770f041f78addaa96e1ef6954732751b57b001950e92865a2f3d830e2R855